Assignments/Midterm.py

```diff
@@ -0,0 +1,687 @@
+    #         "x_2",
+    #         "Bin_Pop",
+    #         "Bin_Mean",
+    #         "Pop_Mean",
+    #         "Pop_Proportion",
+    #         "Mean_Sq_Diff",
+    #         "Mean_Sq_Diff_Weighted",
+    #     ]
+    # )
+    copy_data = data.copy()
+    if is_continuous(data, pred1):
+        copy_data["bins1"] = pd.cut(copy_data[pred1], 10, labels=False)
+        if is_continuous(data, pred2):
+            copy_data["bins2"] = pd.cut(copy_data[pred2], 10, labels=False)
+            calc_matrix = np.zeros((11, 11))
+            for x_1, val1 in enumerate(copy_data["bins1"].unique()):
+                for x_2, val2 in enumerate(copy_data["bins2"].unique()):
+                    bin_mean = copy_data[resp][
+                        (copy_data["bins1"] == x_1) & (copy_data["bins2"] == x_2)
+                    ].mean()
+                    calc_matrix[x_1, x_2] = bin_mean
+        else:
+            calc_matrix = np.zeros((11, len(copy_data[pred2].unique())))
+            for x_1, val1 in enumerate(copy_data["bins1"].unique()):
+                for x_2, val2 in enumerate(copy_data[pred2].unique()):
+                    bin_mean = copy_data[resp][
+                        (copy_data["bins1"] == x_1) & (copy_data[pred2] == val2)
+                    ].mean()
+                    calc_matrix[x_1, x_2] = bin_mean
+    else:
+        if is_continuous(data, pred2):
+            calc_matrix = np.zeros((len(copy_data[pred1].unique()), 11))
+            copy_data["bins2"] = pd.cut(copy_data[pred2], 10, labels=False)
+            for x_1, val1 in enumerate(copy_data[pred1].unique()):
+                for x_2, val2 in enumerate(copy_data["bins2"].unique()):
+                    bin_mean = copy_data[resp][
+                        (copy_data[pred1] == val1) & (copy_data["bins2"] == x_2)
+                    ].mean()
+                    calc_matrix[x_1, x_2] = bin_mean
+        else:
+            calc_matrix = np.zeros(
+                (len(copy_data[pred1].unique()), len(copy_data[pred2].unique()))
+            )
+            for x_1, val1 in enumerate(copy_data[pred1].unique()):
+                for x_2, val2 in enumerate(copy_data[pred2].unique()):
+                    bin_mean = copy_data[resp][
+                        (copy_data[pred1] == val1) & (copy_data[pred2] == val2)
+                    ].mean()
+                    calc_matrix[x_1, x_2] = bin_mean
+    return calc_matrix
+
+
+def make_clickable(url, name):
+    # This function is from stack overflow link that the professor provided in the slides
+    # https://stackoverflow.com/questions/42263946/how-to-create-a-table-with-clickable-hyperlink-in-pandas-jupyter-notebook
+    return '<a href="{}" rel="noopener noreferrer" target="_blank">{}</a>'.format(
+        url, name
+    )
+
+
+def add_plot_links(table, data, resp):
+    path = "/Users/dylanmather/Documents/Grad_School/Fall_21_temp/BDA696_PythonProject2/Assignments/"
+    table["pred1_filepath"] = "placeholder" * len(table.iloc[0])
```

Comment:
I don't think you need the * Len(table stuff. I think if you just set the column it should run.